Assets/MRTK/Services/InputSystem/FocusProvider.cs

```diff
@@ -883,17 +883,14 @@ public bool UnregisterPointer(IMixedRealityPointer pointer)
         /// <inheritdoc />
         public IEnumerable<T> GetPointers<T>() where T : class, IMixedRealityPointer
         {
-            List<T> typePointers = new List<T>();
             foreach (var pointer in pointers.Values)
             {
                 T typePointer = pointer.Pointer as T;
                 if (typePointer != null)
                 {
-                    typePointers.Add(typePointer);
+                    yield return typePointer;
```

Comment:
TIL c# has this kind of lazy evaluation

so cool

Reply:
@RogPodge from C# 7 you can even do this:
```
/// <inheritdoc />
        public IEnumerable<T> GetPointers<T>() where T : class, IMixedRealityPointer
        {
            foreach (var pointer in pointers.Values)
            {
                if (pointer.Pointer as T typePointer)
                {
                    yield return typePointer;
                }
            }
        }
```